File: editorial/chap06/python/answer_B41.py
# 입력
X, Y = map(int, input().split())

# 거꾸로 생각하기
Answer = []
while X>=2 or Y>=2:
	Answer.append([X, Y])
	if X > Y:
		X -= Y
	else:
		Y -= X
Answer.reverse()

# 출력
print(len(Answer))
for i in range(len(Answer)):
	print(str(Answer[i][0]) + ' ' + str(Answer[i][1]))

File: editorial/chap06/cpp/answer_B41.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;
 
int X, Y;
 
int main() {
	// 입력
	cin >> X >> Y;
 
	// 거꾸로 생각하기
	vector<pair<int, int>> Answer;
	while (X >= 2 || Y >= 2) {
		Answer.push_back(make_pair(X, Y));
		if (X > Y) X -= Y;
		else Y -= X;
	}
	reverse(Answer.begin(), Answer.end());
 
	// 출력
	cout << Answer.size() << endl;
	for (int i = 0; i < Answer.size(); i++) {
		cout << Answer[i].first << " " << Answer[i].second << endl;
	}
	return 0;
}
